pages/project-version/middleware/index.js

```diff
@@ -183,14 +183,20 @@ const getPreviousVersion = (req, type = 'project-versions') => {
 };
 
 const getGrantedVersion = (req, type = 'project-versions') => {
-  const key = type === 'project-versions' ? 'granted' : 'grantedRa';
-  if (!req.project || !req.project[key]) {
+  if (!req.project) {
+    return Promise.resolve();
+  }
+  const key = type === 'project-versions' ? 'versions' : 'retrospectiveAssessments';
+  const model = type === 'project-versions' ? 'version' : 'retrospectiveAssessment';
+
+  const granted = req.project[key]
```

Comment:
Can we be sure `req.project[key]` always exists here?

Reply:
Probably, but I've added a check just in case.